### PR TITLE
Support for configuration via environment config

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Depends on: https://github.com/topaxi/ember-service-worker-update-notify/pull/16
Mostly from: https://github.com/topaxi/ember-service-worker-update-notify/pull/13

Thanks, @velrest !

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,13 @@ The time is measured in milliseconds. The default is 1200000 (20 minutes).
 </ServiceWorkerUpdateNotify>
 ```
 
+The poll interval can also be configured in your `.ember-cli-buil.js` with:
```
Comment (jelhan) [inline README.md]: It's configured through `config/environment.js`, isn't it?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,13 @@ The time is measured in milliseconds. The default is 1200000 (20 minutes).
...
+```js
+'ember-service-worker-update-notify': {
+  pollingInterval:1200000 // Default is 20min
+}
+```
```
Comment (jelhan) [inline README.md]: Would be helpful if some more context of the surrounding file is present:

```js
// config/environment.js

module.exports = function(environment) {
  let ENV = {
    'ember-service-worker-update-notify': {
      pollingInterval:1200000 // Default is 20min
    }
  };
 
 return ENV;
};
```

File: addon/components/service-worker-update-notify.js [hunk 3]
```diff
@@ -27,45 +31,51 @@ export default Component.extend({
...
-      // Also, the tasks use a while(true) loop,
-      // which would prevent tests from every finishing
-      this.setupTask.perform();
+    // NOTE: this.attrs *shouldn't* be used
+    //       as it's poorly documented.
+    //       Consider moving to a glimmer component
+    //       where this.args are available.
+    //
+    // eslint-disable-next-line ember/no-attrs-in-components
+    let wasPassedIn = Boolean(this.attrs.pollingInterval);
```
Comment (jelhan) [inline addon/components/service-worker-update-notify.js]: Is `hasOwnProperty()` not working for `this.attrs`? Would be easier to read IMO.

File: README.md (hunk 1 above)
Comment (NullVoxPopuli) [reply README.md]: Whoops. Yup

File: addon/components/service-worker-update-notify.js (hunk 3 above)
Comment (NullVoxPopuli) [reply addon/components/service-worker-update-notify.js]: Oh, it probably would, I didn't think of that. Thanks!

**Comment by simonihmig** [comment]
@NullVoxPopuli have you noticed this is failing tests for Ember 3.8 and lower? Seems `getOwner()` does not return anything. No idea why...  🤔

**Comment by NullVoxPopuli** [comment]
yeah, that's super weird.

and the upcoming changes are cool. I just want to get this config option re-implemented, since I had it reverted a while ago, due to other breaking changes :(

File: addon/components/service-worker-update-notify.js [hunk 4]
```diff
@@ -27,45 +32,51 @@ export default Component.extend({
   // public
   pollingInterval: 1200000, // 20 minutes in ms
 
-  didInsertElement() {
+  config: computed(function() {
```
Comment (simonihmig) [inline addon/components/service-worker-update-notify.js]: Right! That *should* fix the tests I believe. Good catch!

Comment (NullVoxPopuli) [reply addon/components/service-worker-update-notify.js]: I'm a little sad the native getter isn't supported in <= ember 3.8 :(

Comment (NullVoxPopuli) [reply addon/components/service-worker-update-notify.js]: cause this is like.. forced memoization

**Pull request merged**